vpn/bytes.go

```diff
@@ -113,6 +113,9 @@ func bytesUnpadPKCS7(b []byte, blockSize int) ([]byte, error) {
 
 // bytesPadPKCS7 returns the PKCS#7 padding of a byte array.
 func bytesPadPKCS7(b []byte, blockSize int) ([]byte, error) {
+	if blockSize == 0 {
+		return nil, fmt.Errorf("%w: %s", errBadInput, "blocksize cannot be zero")
```

Comment:
* [x] MIV-01-002 Possible DoS via Integer Division by Zero

Reply:
While we are at it, I think we should also protect against negative blockSize values. The assumption in the report is that the attacker may be able to control the `blockSize` value, hence if that is the case, this snippet will also lead to a crash:
```
    b := []byte("a")
    blockSize := -1
    psiz := blockSize - len(b) % blockSize
    padding := bytes.Repeat([]byte{byte(psiz)}, psiz)
```

via

```
panic: bytes: negative Repeat count

goroutine 1 [running]:
bytes.Repeat(0xc0000a2f27, 0x1, 0x1, 0xffffffffffffffff, 0x46, 0x4db500, 0xc000010080)
    /usr/local/go/src/bytes/bytes.go:499 +0x1aa
main.main()
    /home/runner/3bh1srnac6d/main.go:11 +0x62
```